nodejs/eks/cluster.ts

```diff
@@ -579,9 +596,68 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         nodeSecurityGroupTags: args.nodeSecurityGroupTags,
         storageClasses: userStorageClasses,
         fargateProfile: fargateProfile,
+        oidcProvider: oidcProvider,
     };
 }
 
+// Adapted from https://git.io/JvGHB
+
+// Find the root CA cert in a chain of certs.
+function findCACertificate(certificate: any, parent?: any): any {
+    if (certificate?.issuerCertificate.fingerprint !== certificate.fingerprint) {
```

Comment:
I actually don't know `?.` well - is this "safe"?  Will the undefined propagated by the `?.` flow into the following `.`?

Reply:
Looks like you would actually need to do `certificate?.issuerCertificate?.fingerprint` for this to work properly.

See https://www.typescriptlang.org/v2/en/play#code/C4TwDgpgBAQghgJygXigbwFBW1ARogLigGdgEBLAOwHMMBfDUSKAMQHs2V0scAzDgPxF4CehgDGbSqSj82Rdp1RoGk6cDyIucgQDodu-AgDcEqcTYAbCLsttqACiMBKIA for an example.